backend/src/main/java/com/redhat/cloud/notifications/routers/internal/SourcesSecretsMigrationService.java

```diff
@@ -40,34 +52,105 @@ public class SourcesSecretsMigrationService {
     @Deprecated(forRemoval = true)
     @POST
     @Path("/sources-migration")
-    @Transactional
     public void migrateEndpointSecretsSources() {
-        final List<Endpoint> endpoints = this.endpointRepository.findEndpointWithPropertiesWithStoredSecrets();
+        final Map<UUID, String> eligibleEndpoints = this.endpointRepository.findEndpointWithPropertiesWithStoredSecrets();
 
         int processedCounter = 0;
         int errorCounter = 0;
-        for (final Endpoint endpoint : endpoints) {
+        for (final Map.Entry<UUID, String> eligibleEndpoint : eligibleEndpoints.entrySet()) {
             // On failure, don't stop processing the endpoints and keep going,
             // since this is not a destructive operation.
             try {
-                this.secretUtils.createSecretsForEndpoint(endpoint);
+                migrateEndpoint(eligibleEndpoint);
             } catch (final Exception e) {
-                if (e instanceof WebApplicationException wae) {
-                    Log.errorf("[endpoint_id: %s] error when migrating the endpoint secrets: unable to create the secrets for the endpoint: %s", endpoint.getId(), wae.getResponse().getEntity());
+                if (this.extractConstraintViolationException(e) instanceof ConstraintViolationException cve) {
+                    Log.errorf("[endpoint_id: %s] error when migrating the endpoint secrets: a constraint violation exception was raised: %s. Constraints list: %s", eligibleEndpoint.getKey(), cve.getMessage(), cve.getConstraintViolations());
+                } else if (e instanceof NotFoundException) {
+                    Log.errorf("[endpoint_id: %s] error when migrating the endpoint secrets: unable to find endpoint in the database", eligibleEndpoint.getKey());
+                    // The endpoint was not found, so we need to continue the
+                    // loop since there is nothing to delete.
+                    continue;
+                } else if (e instanceof WebApplicationException wae) {
+                    Log.errorf("[endpoint_id: %s] error when migrating the endpoint secrets: unable to create the secrets for the endpoint: %s", eligibleEndpoint.getKey(), wae.getResponse().getEntity());
+                } else {
+                    Log.errorf("[endpoint_id: %s] error when migrating the endpoint secrets: unable to create the secrets for the endpoint: %s", eligibleEndpoint.getKey(), e.getMessage());
+                }
+
+                // Delete the secrets for the processed endpoint, since an
+                // error occurred.
+                final Endpoint endpoint = this.processedEndpoints.get(eligibleEndpoint.getKey());
```

Comment:
Will this ever return an endpoint? Line 123 (the `put` call) will only be executed if no exception is thrown when `migrateEndpoint` is run.

Reply:
Mmmmm... You are right, I think the only time this will ever return an endpoint is if we hit a `ConstraintViolationException`, since the `put` call will be executed and then, when attempting to save the endpoint, the exception will be thrown... So I guess I can move the code up there, since if there is a `NotFoundException` then the secrets won't be created, and if we have a `WebApplicationException`, that simply means that Sources didn't like the payload and didn't create the secrets...